lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -0,0 +1,316 @@
+module Msf::Payload::Adapter::Fetch
+
+  def initialize(*args)
+    super
+    register_options(
+      [
+        Msf::OptEnum.new('FETCH_COMMAND', [true, 'Command to fetch payload', 'CURL', %w{ WGET CURL FTP TNFTP TFTP Certutil Psh_InvokeWebRequest }]),
+        Msf::OptPort.new('FETCH_SRVPORT', [true, 'Local port to use for serving payload', 8080]),
+        Msf::OptAddressLocal.new('FETCH_SRVHOST', [ true, 'Local IP to use for serving payload', "0.0.0.0"]),
+        Msf::OptString.new('FETCH_URIPATH', [ false, 'Local URI to use for serving payload', '']),
+        Msf::OptString.new('FETCH_WRITABLE_DIR', [ true, 'Remote writable dir to store payload; cannot contain spaces.', ''], regex:/^[\S]*$/),
```

Comment:
Not sure if it was already discussed, but does a default of `/tmp` make sense here? 👀

Reply:
Good point- I'd avoided a default value here because it had to support both Windows and Linux, but that was before we created the separate Windows/Linux options files for `FETCH_COMMAND`.  It would make sense to move the `FETCH_WRITABLE_DIR` out of `fetch.rb` and into the OS-specific options now that we have them.  Linux could be `/tmp`, and windows could be `%TEMP%`.

Reply:
After thinking about it, we should do the same with `FETCH_FILENAME` so we can more transparently enforce the `.exe` extension that Windows needs.

Reply:
I also think options should be set to the more relevant value by default. Maybe the option can be unset for TFTP payloads?